src/VisualStudio/CSharp/Impl/Debugging/DataTipInfoGetter.cs

```diff
@@ -20,7 +20,11 @@ internal static class DataTipInfoGetter
         {
             try
             {
-                var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                // Try to fetch the syntax tree synchronously. We are blocking the UI thread when we are doing this,
```

Comment:
if we believe doing synchronously is better when we are on UI thread, shouldn't we do same for GetSemanticModel? and change this method sync method rather than async?